jraft-core/src/main/java/com/alipay/sofa/jraft/core/Replicator.java

```diff
@@ -610,26 +614,27 @@ private void startHeartbeatTimer(final long startMs) {
             this.heartbeatTimer = this.timerManager.schedule(() -> onTimeout(this.id), dueTime - Utils.nowMs(),
                 TimeUnit.MILLISECONDS);
         } catch (final Exception e) {
-            LOG.error("Fail to schedule heartbeat timer", e);
+            LOG.error("Replicator {} fail to schedule heartbeat timer ", this, e);
             onTimeout(this.id);
         }
     }
 
     void installSnapshot() {
         if (getState() == State.Snapshot) {
-            LOG.warn("Replicator {} is installing snapshot, ignore the new request.", this.options.getPeerId());
+            LOG.warn("Replicator {} is installing snapshot, ignore the new request.", this);
             unlockId();
             return;
         }
         boolean doUnlock = true;
         if (!this.rpcService.connect(this.options.getPeerId().getEndpoint())) {
-            LOG.error("Fail to check install snapshot connection to peer={}, give up to send install snapshot request.", this.options.getPeerId().getEndpoint());
+            LOG.error("Fail to check install snapshot connection to peer={}, give up to send install snapshot request, group id {}.",
+                    this.options.getPeerId().getEndpoint(), this.options.getGroupId());
             block(Utils.nowMs(), RaftError.EHOSTDOWN.getNumber());
             return;
         }
         try {
             Requires.requireTrue(this.reader == null,
-                "Replicator %s already has a snapshot reader, current state is %s", this.options.getPeerId(),
+                "Replicator %s already has a snapshot reader, current state is %s", this.toString(),
```

Comment:
Perhaps `(peerId , groupId, replicatorType)` are enough, the information in `this.toString` is too complicated.